torchbenchmark/__init__.py

```diff
@@ -103,6 +111,230 @@ def setup(verbose: bool = True, continue_on_fail: bool = False) -> bool:
     return len(failures) == 0
 
 
+@dataclasses.dataclass(frozen=True)
+class ModelDetails:
+    """Static description of what a particular TorchBench model supports.
+
+    When parameterizing tests, we only want to generate sensible ones.
+    (e.g. Those where a model can be imported and supports the feature to be
+    tested or benchmarked.) This requires us to import the model; however many
+    of the models are EXTREMELY stateful, and even importing them consumes
+    significant system resources. As a result, we only want one (or a few)
+    alive at any given time.
+
+    This class describes what a particular model does and does not support, so
+    that we can release the underlying subprocess but retain any pertinent
+    metadata.
+    """
+    path: str
+    exists: bool
+    optimized_for_inference: bool
+    _diagnostic_msg: str
+
+    @property
+    def name(self) -> str:
+        return os.path.basename(self.path)
+
+
+class ModelTask(base_task.TaskBase):
+
+    # The worker may (and often does) consume significant system resources.
```

Comment:
I am curious about this approach of running multiple torchbench tasks at the same time. There is fear in me that it would interfere with the results. But there is no data I have to support this fear - not that emotions need be rational :)

Can you share more as to the impact you think parallelization of benchmarks might have? Do we have any early numbers?

Reply:
It's mostly more for the context of ubenchmarking, where you're mostly running single core (or maybe a couple cores) and you can strongly isolate the work units. For full model benchmarks it's less likely to be applicable. And as you say, there are resource contention concerns.